### PR TITLE
platform dir: fixed some javadoc discrepancies

### PR DESCRIPTION
**Pull request by tbw777**
Some discrepancies are from refactoring

### CODE REVIEW

File: platform/api.io/src/org/netbeans/api/io/IOProvider.java [hunk 1]
```diff
@@ -214,7 +214,8 @@ public abstract InputOutput getIO(@NonNull String name, boolean newIO,
      *
      * @param <IO>
      * @param <OW>
-     * @param <POS>
+     * @param <P>
+     * @param <F>
```
Comment (mbien) [inline platform/api.io/src/org/netbeans/api/io/IOProvider.java:218]: tags without doc are useless. Either they get a description or they can be all removed.

File: platform/api.search/src/org/netbeans/api/search/provider/impl/SimpleSearchIterator.java [hunk 2]
```diff
@@ -220,7 +218,7 @@ public void remove() {
 
     /**
      * Sort enumeration by fila names;
-     *
+     * @param enm
```
Comment (mbien) [inline platform/api.search/src/org/netbeans/api/search/provider/impl/SimpleSearchIterator.java:221]: can be removed

File: platform/autoupdate.services/src/org/netbeans/modules/autoupdate/services/Utilities.java [hunk 3]
```diff
@@ -142,6 +142,9 @@ public static Collection<KeyStore> getKeyStore (TrustLevel trustLevel) {
      *
      * @param archiveCertPath
      * @param trustedCertificates
+     * @param trustedCACertificates
+     * @param validateCertificates
+     * @param validationCACertificates
      * @return
      */
```
Comment (mbien) [inline platform/autoupdate.services/src/org/netbeans/modules/autoupdate/services/Utilities.java:149]: same here. The whole comment serves no purpose.

File: platform/api.io/src/org/netbeans/api/io/IOProvider.java (hunk 1 above)
Comment (tbw777) [reply platform/api.io/src/org/netbeans/api/io/IOProvider.java]: fixed below

File: platform/api.search/src/org/netbeans/api/search/provider/impl/SimpleSearchIterator.java (hunk 2 above)
Comment (tbw777) [reply platform/api.search/src/org/netbeans/api/search/provider/impl/SimpleSearchIterator.java]: fixed below

File: platform/autoupdate.services/src/org/netbeans/modules/autoupdate/services/Utilities.java (hunk 3 above)
Comment (tbw777) [reply platform/autoupdate.services/src/org/netbeans/modules/autoupdate/services/Utilities.java]: fixed below

**Comment by ebarboni** [comment]
seems ok to me, @mbien ?

**Comment by mbien** [comment]
@ebarboni ok to me too. I looked through it only once and not in detail.

**Comment by ebarboni** [comment]
merging

**Pull request merged**